### PR TITLE
send min/median connection score along with avg

### PR DESCRIPTION
**Pull request by shishirng**

### CODE REVIEW

File: pkg/utils/math.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	num := len(input)
+	if num == 0 {
+		return 0
+	}
+	sort.Slice(input, func(i, j int) bool {
+		return input[i] < input[j]
+	})
+	if num%2 != 0 {
+		return input[num/2]
+	} else {
```
Comment (boks1971) [inline pkg/utils/math.go:16]: guess this else can be the final path. Don't need the 0 return at the end.

**Review by boks1971 — APPROVED** [review]

**Review by davidzhao — APPROVED** [review]
lg

File: pkg/telemetry/statsworker.go [hunk 2]
```diff
@@ -154,7 +157,15 @@ func coalesce(stats []*livekit.AnalyticsStat) *livekit.AnalyticsStat {
 			continue
 		}
 
-		score += stat.Score
+		// only consider non-zero scores
+		if stat.Score > 0 {
```
Comment (noahlt) [inline pkg/telemetry/statsworker.go:161]: what does a zero score mean?

File: pkg/utils/math.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	if num == 0 {
+		return 0
+	}
+	sort.Slice(input, func(i, j int) bool {
+		return input[i] < input[j]
+	})
+	if num%2 != 0 {
+		return input[num/2]
+	}
+	left := input[(num-1)/2]
```
Comment (noahlt) [inline pkg/utils/math.go:17]: I am pretty sure this should be:

```go
left := input[num/2 - 1]
right := input[num/2]
```

And if we do that we need another early return:

```go
if num == 0 {
  return 0
} else if num == 1 {
  return input[num]
}
```

The current behavior [fails on odd numbers](https://go.dev/play/p/DtscbuDRSCK)

**Review by noahlt — CHANGES_REQUESTED** [review]
see median comments

File: pkg/telemetry/statsworker.go (hunk 2 above)
Comment (boks1971) [reply pkg/telemetry/statsworker.go:161]: I believe it is just a sanity check. The scorer does not produce 0 (in theory it should not produce 0, but there could always be bugs😊)

File: pkg/utils/math.go (hunk 3 above)
Comment (boks1971) [reply pkg/utils/math.go:17]: The odd number case is covered above. So both works (what Shishir had and what you suggested) for even numbers.

File: pkg/utils/math.go [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+import "sort"
+
+// MedianFloat32 gets median value for an array of float32
+func MedianFloat32(input []float32) float32 {
+	num := len(input)
+	if num == 0 {
+		return 0
+	} else if num == 1 {
+		return input[num]
```
[the remark this answers is not in the document]
Comment (noahlt) [reply pkg/utils/math.go:11]: Right >_<

boy a real team effort here

**Review by noahlt — APPROVED** [review]

**Pull request merged**